cli-plugins/manager/cobra.go

```diff
@@ -90,11 +90,18 @@ func AddPluginCommandStubs(dockerCli command.Cli, rootCmd *cobra.Command) (err e
 					cargs = append(cargs, args...)
 					cargs = append(cargs, toComplete)
 					os.Args = cargs
-					runCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
+					pluginRunCommand, runErr := PluginRunCommand(dockerCli, p.Name, cmd)
 					if runErr != nil {
 						return nil, cobra.ShellCompDirectiveError
 					}
-					runErr = runCommand.Run()
+					if verifiedDockerCli, ok := dockerCli.(*command.DockerCli); ok {
```

Comment:
Can we just shadow `dockerCli` in this case instead of calling it `verifiedDockerCli`? Not a huge fan, sorry (I was wondering what made it "verified", and then realized it was just the typecast cli)

Reply:
thing is if we shadow `dockerCli` here, then in the else condition we can't call `dockerCli.Err()` as it'd be a zero value and not the original value of `dockerCli`. I tried avoiding this where possible (see the other places where i do this cast), but here since we need the `else` i wasn't sure how to proceed if not renaming the var. 

Maybe there are better ways?

Reply:
Could we add a (can be un-exported) `pluginInstrumenter` interface and match that?